### ISSUE TITLE
[clang] Captured structured binding is wrongly considered a bitfield

### ISSUE THREAD

**Issue by Rajveer100** [llvm/llvm-project#82918]
```C++
struct S {
    unsigned short i:2;
};

int main() {
    auto [i] = S();
    (void) [i]() mutable { i = 42; };
}
```

In the above snippet, assignment `i = 42` produces a false warning:

```
warning: implicit truncation from 'int' to bit-field changes value from 42 to 2 [-Wbitfield-constant-conversion]
    6 |     (void) [i]() mutable { i = 42; };
      |                              ^ ~~
```

The `i` being assigned is the captured binding as the lambda function's parameter.

[GodBolt](https://godbolt.org/z/dTbneKG46)

**Comment by shafik** [comment]
CC @cor3ntin not sure what the behavior should be here: https://reviews.llvm.org/D122768?id=419880
